src/SurveyService.Validation/Option/CreateOptionRequestValidator.cs

```diff
@@ -0,0 +1,42 @@
+﻿using FluentValidation;
+using LT.DigitalOffice.SurveyService.Data.Interfaces;
+using LT.DigitalOffice.SurveyService.Models.Dto.Requests.Option;
+using LT.DigitalOffice.SurveyService.Validation.Option.Interfaces;
+
+namespace LT.DigitalOffice.SurveyService.Validation.Option;
+
+public class CreateOptionRequestValidator : AbstractValidator<(CreateOptionRequest request, bool? hasCustomOptions)>, ICreateOptionRequestValidator
+{
+  private readonly IQuestionRepository _questionRepository;
+
+  public CreateOptionRequestValidator(
+    IQuestionRepository questionRepository)
+  {
+    _questionRepository = questionRepository;
+
+    When(x => x.hasCustomOptions is null, () =>
+    {
+      RuleFor(x => x.request.QuestionId)
+        .MustAsync(async (x, _) => await _questionRepository.GetAsync(x) is not null)
```

Comment:
For this check usually add to repository method DoesExistAsync, which receives Id and returns bool

Reply:
Done